addons/binding/org.openhab.binding.jeelink/src/main/java/org/openhab/binding/jeelink/handler/ec3k/Ec3kReadingConverter.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.jeelink.handler.ec3k;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.openhab.binding.jeelink.handler.JeeLinkReadingConverter;
+
+/**
+ * Converter for converting a line read from a ec3kSerial sketch to a Ec3kReading.
+ *
+ * @author Volker Bier - Initial contribution
+ */
+public class Ec3kReadingConverter extends JeeLinkReadingConverter<Ec3kReading> {
+    private static final Pattern LINE_P = Pattern
+            .compile("OK\\s+22\\s+([0-9]+)\\s+([0-9]+)\\s+([0-9]+)\\s+([0-9]+)\\s+([0-9]+)\\s+([0-9]+)\\s+([0-9]+)"
+                    + "\\s+([0-9]+)\\s+([0-9]+)\\s+([0-9]+)\\s+([0-9]+)\\s+([0-9]+)\\s+([0-9]+)\\s+([0-9]+)\\s+([0-9]+)"
+                    + "\\s+([0-9]+)\\s+([0-9]+)\\s+([0-9]+)\\s+([0-9]+)\\s+([0-9]+)");
+
+    @Override
+    protected Ec3kReading createReading(String inputLine) {
+        if (listenerIdMap.size() > 0 && inputLine != null) {
+            Matcher matcher = LINE_P.matcher(inputLine);
+            if (matcher.matches()) {
+                /*
+                 * [ec3kSerial.1]
```

Comment:
Does not really belong here. Is part of the header that is written by the sketch running on the USB stick.